File: editor/src/de/fatox/meta/ui/FolderListAdapter.java
package de.fatox.meta.ui;

import com.badlogic.gdx.utils.Array;
import com.kotcrab.vis.ui.VisUI;
import com.kotcrab.vis.ui.util.adapter.ArrayAdapter;
import com.kotcrab.vis.ui.util.adapter.SimpleListAdapter;
import com.kotcrab.vis.ui.widget.VisTable;
import de.fatox.meta.ui.components.MetaTextButton;

public class FolderListAdapter<ItemT> extends ArrayAdapter<ItemT, VisTable> {
	private final SimpleListAdapter.SimpleListAdapterStyle style;

	public FolderListAdapter(Array<ItemT> array) {
		super(array);
		style = VisUI.getSkin().get("default", SimpleListAdapter.SimpleListAdapterStyle.class);
	}


	@Override
	protected VisTable createView(ItemT item) {
		VisTable table = new VisTable();
		table.pad(1);
		MetaTextButton visTextButton = new MetaTextButton(item.toString(), 12);
		table.add(visTextButton).growX().prefWidth(128).pad(1);
		return table;
	}

	@Override
	protected void selectView(VisTable view) {
		view.setBackground(style.selection);
	}

	@Override
	protected void deselectView(VisTable view) {
		view.setBackground(style.background);
	}
}

File: editor/src/main/kotlin/de/fatox/meta/ui/FolderListAdapter.kt
package de.fatox.meta.ui

import com.badlogic.gdx.scenes.scene2d.Actor
import com.badlogic.gdx.scenes.scene2d.utils.Drawable
import com.badlogic.gdx.utils.Array
import de.fatox.meta.ui.MetaSkin
import de.fatox.meta.ui.components.MetaArrayAdapter
import de.fatox.meta.ui.components.MetaTable
import de.fatox.meta.ui.components.MetaTextButton

class FolderListAdapter<ItemT>(array: Array<ItemT>?) : MetaArrayAdapter<ItemT, MetaTable>(array) {
    private val background: Drawable = MetaSkin.skin().getDrawable("meta.panel")
    private val selection: Drawable = MetaSkin.skin().getDrawable("meta.selection")

    override fun createView(item: ItemT): MetaTable {
        val table = MetaTable()
        table.pad(1f)
        val visTextButton = MetaTextButton(item.toString(), 12)
        table.add(visTextButton).growX().prefWidth(128f).pad(1f)
        return table
    }

    override fun selectView(view: Actor) {
        (view as? MetaTable)?.background = selection
    }

    override fun deselectView(view: Actor) {
        (view as? MetaTable)?.background = background
    }

}
